### PR TITLE
enhance show parts and show hosts

### LINKED ISSUE #1226: Support list the partition specified in console
**Is your feature request related to a problem? Please describe.**
It 's maybe a subtask of the PR #1086, that from the [discussion](https://github.com/vesoft-inc/nebula/pull/1086#discussion_r341891011).

**Describe the solution you'd like**

1. The specified partition ID as the parameter.
        (user@ip) [space]> SHOW PART **666**;
2. The result is sorted by partition IDs, so it maybe use the **LIMIT** clause easily.
        (user@ip) [space]>  SHOW PARTS **| LIMIT 666, 1**
3.  Both of the above are implemented.

Which solution should we choose?

### PR DESCRIPTION
**Pull request by critical27**
1. Add `show part $id` and `show parts $id1, $id2, ..., $idn`
image

image

    close #1226 

2. Enhance `show hosts`, we can show total leader count
image

### CODE REVIEW

File: src/parser/parser.yy [hunk 1]
```diff
@@ -1438,6 +1438,9 @@ show_sentence
     | KW_SHOW KW_PARTS {
         $$ = new ShowSentence(ShowSentence::ShowType::kShowParts);
     }
+    | KW_SHOW KW_PARTS INTEGER {
```
Comment (bright-starry-sky) [inline src/parser/parser.yy]: `KW_PARTS` ? or `KW_PART`?

Comment (critical27) [reply src/parser/parser.yy]: Maybe both? It seems most of command in console is `xxxxs`.

Comment (bright-starry-sky) [reply src/parser/parser.yy]: Aha，A difficult choice 😄

**Comment by Amber1990Zhang** [comment]
Thank you for taking care of this. Will update the docs once the code is ready.

File: src/parser/test/ParserTest.cpp [hunk 2]
```diff
@@ -948,6 +948,12 @@ TEST(Parser, AdminOperation) {
         auto result = parser.parse(query);
         ASSERT_TRUE(result.ok()) << result.status();
     }
+    {
+        GQLParser parser;
+        std::string query = "SHOW PARTS 666";
```
Comment (zhangguoqing) [inline src/parser/test/ParserTest.cpp]: How about ```SHOW PART 666;``` or ```SHOW PARTS 111,222,333,444,555,666;```?

Comment (Amber1990Zhang) [inline src/parser/test/ParserTest.cpp]: If you're showing one part, it's better to use `SHOW PART 666`.

Comment (critical27) [reply src/parser/test/ParserTest.cpp]: Good suggestion

Comment (critical27) [reply src/parser/test/ParserTest.cpp]: I'll inform you after merged, you can hold on doc.

File: src/parser/parser.yy (hunk 1 above)
Comment (critical27) [reply src/parser/parser.yy]: both supported for now

File: src/parser/AdminSentences.h [hunk 3]
```diff
@@ -332,6 +344,20 @@ class HostList final {
     std::vector<std::unique_ptr<HostAddr>>      hosts_;
 };
 
+class IntegerList final {
```
Comment (dangleptr) [inline src/parser/AdminSentences.h]: Can't we use std::vector directly ?

Comment (critical27) [reply src/parser/AdminSentences.h]: Good point

File: src/graph/ShowExecutor.cpp [hunk 4]
```diff
@@ -151,6 +155,41 @@ void ShowExecutor::showHosts() {
...
+        {
+            // set sum of leader/partitions of all hosts
+            std::vector<cpp2::ColumnValue> row;
+            row.resize(6);
+
+            int32_t leaderCount = 0;
+            std::string leaders;
+            for (const auto& spaceEntry : leaderPartsCount) {
+                leaders += spaceEntry.first + ": " +
+                           folly::to<std::string>(spaceEntry.second) + ", ";
```
Comment (bright-starry-sky) [inline src/graph/ShowExecutor.cpp]: Is there an extra comma at the end of the line?

File: src/graph/ShowExecutor.cpp [hunk 5]
```diff
@@ -151,6 +155,41 @@ void ShowExecutor::showHosts() {
...
+                leaderCount += spaceEntry.second;
+            }
+            if (!leaders.empty()) {
+                leaders.resize(leaders.size() - 2);
+            }
+
+            std::string parts;
+            for (const auto& spaceEntry : allPartsCount) {
+                parts += spaceEntry.first + ": " +
+                         folly::to<std::string>(spaceEntry.second) + ", ";
```
Comment (bright-starry-sky) [inline src/graph/ShowExecutor.cpp]: ditto

File: src/graph/ShowExecutor.cpp (hunk 4 above)
Comment (critical27) [reply src/graph/ShowExecutor.cpp]: Good catch

File: src/parser/AdminSentences.h (hunk 3 above)
Comment (dangleptr) [inline src/parser/AdminSentences.h]: Do we still need this class now?

Comment (critical27) [reply src/parser/AdminSentences.h]: Fixed, 💯

**Pull request merged**